multi.c

```diff
@@ -0,0 +1,145 @@
+#include "multi.h"
+
+#include "brute.h"
+#include "queue.h"
+#include "single.h"
+#include <assert.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+void *
+mt_password_check (void *context)
+{
+  mt_context_t *mt_context = (mt_context_t *)context;
+  task_t task;
+  st_context_t st_context = {
+    .hash = mt_context->config->hash,
+    .data = { .initialized = 0 },
+  };
+
+  while (true)
+    {
+      /* don't understand why we need status check here */
```

Comment:
That's how you get out of infinite loop when all tasks are complete. queue_cancel will force queue_pop to return a error status